### PR TITLE
[FSSDK-10761] feat: make vuid as opt-in

### PR DESCRIPTION
**Pull request by muzahidul-opti**
## Summary
- Make vuid feature as opt-in.
- Reference PR: https://github.com/optimizely/javascript-sdk/pull/950/files
 ## Logical changes when vuid is not enabled 
- vuid is not generated/saved.
- Delete any existing vuid (leftover by previous init).
- “client-intialized” event not auto-fired.
- vuid is not included in ODP events as a default attribuite for tracking. 
- createUserContext() will be rejected when userId is not provided.

## Issues
- [FSSDK-10761](https://jira.sso.episerver.net/browse/FSSDK-10761)

### CODE REVIEW

File: Sources/ODP/OdpVuidManager.swift [hunk 1]
```diff
@@ -17,14 +17,17 @@
 import Foundation
 
 class OdpVuidManager {
-    var vuid: String = ""
+    private var _vuid: String = ""
+    private(set) var enabled: Bool
     let logger = OPTLoggerFactory.getLogger()
-
-    // a single vuid should be shared for all SDK instances
-    static let shared = OdpVuidManager()
```
Comment (jaeopt) [inline Sources/ODP/OdpVuidManager.swift:24]: We should keep it a singleton, so can share a single vuid per device.

File: Sources/ODP/OdpVuidManager.swift [hunk 2]
```diff
@@ -45,6 +48,14 @@ class OdpVuidManager {
 // MARK: - VUID Store
 
 extension OdpVuidManager {
+    var vuid: String {
+        if enabled {
+            return _vuid
+        } else {
+            logger.w("VUID is not enabled.")
+            return ""
```
Comment (jaeopt) [inline Sources/ODP/OdpVuidManager.swift:56]: consistently vuid=nil if not enabled?

File: Sources/ODP/OptimizelySdkSettings.swift [hunk 3]
```diff
@@ -43,13 +45,15 @@ public struct OptimizelySdkSettings {
...
                 disableOdp: Bool = false,
+                enabledVuid: Bool = false,
                 sdkName: String? = nil,
                 sdkVersion: String? = nil) {
         self.segmentsCacheSize = segmentsCacheSize
         self.segmentsCacheTimeoutInSecs = segmentsCacheTimeoutInSecs
         self.timeoutForSegmentFetchInSecs = timeoutForSegmentFetchInSecs
         self.timeoutForOdpEventInSecs = timeoutForOdpEventInSecs
         self.disableOdp = disableOdp
+        self.enableVuid = enabledVuid
```
Comment (jaeopt) [inline Sources/ODP/OptimizelySdkSettings.swift:56]:
```suggestion
        self.enableVuid = enableVuid
```

File: Sources/Optimizely/OptimizelyClient.swift [hunk 4]
```diff
@@ -60,6 +60,7 @@ open class OptimizelyClient: NSObject {
     var decisionService: OPTDecisionService!
     public var notificationCenter: OPTNotificationCenter?
     public var odpManager: OdpManager!
+    private var vuidManager: OdpVuidManager!
```
Comment (jaeopt) [inline Sources/Optimizely/OptimizelyClient.swift:63]:
```suggestion
    private var vuidManager: VuidManager!
```

File: Sources/Optimizely/OptimizelyClient.swift [hunk 5]
```diff
@@ -972,7 +975,11 @@ extension OptimizelyClient {
     
     /// the device vuid (read only)
     public var vuid: String {
```
Comment (jaeopt) [inline Sources/Optimizely/OptimizelyClient.swift:977]:
```suggestion
    public var vuid: String? {
```

**Review by jaeopt — CHANGES_REQUESTED** [review]
Looks good. A few suggestions.

File: Sources/ODP/OptimizelySdkSettings.swift (hunk 3 above)
Comment (muzahidul-opti) [reply Sources/ODP/OptimizelySdkSettings.swift:56]: Done

File: Sources/Optimizely/OptimizelyClient.swift (hunk 5 above)
Comment (muzahidul-opti) [reply Sources/Optimizely/OptimizelyClient.swift:977]: Done

File: Sources/ODP/OdpVuidManager.swift [hunk 6]
```diff
@@ -17,14 +17,17 @@
 import Foundation
 
 class OdpVuidManager {
```
[the remark this answers is not in the document]
Comment (muzahidul-opti) [reply Sources/ODP/OdpVuidManager.swift:19]: Renamed as `VuidManager` and move to Optimizely folder

File: Sources/ODP/OdpManager.swift [hunk 7]
```diff
@@ -97,15 +94,13 @@ public class OdpManager {
...
         }
 
-        var vuid = vuidManager.vuid
-        var fsUserId: String? = userId
         if OdpVuidManager.isVuid(userId) {
             // overwrite if userId is vuid (when userContext is created with vuid)
-            vuid = userId
-            fsUserId = nil
+            eventManager.identifyUser(vuid: userId, userId: nil)
+        } else if let _vuid = vuid {
```
[the remark this answers is not in the document]
Comment (muzahidul-opti) [reply Sources/ODP/OdpManager.swift]: Yah, test case is failing. Made the vuid optional to `identifyUser`.

```
func identifyUser(vuid: String?, userId: String?) {
        var identifiers = [String: String]()
        if let _vuid = vuid {
            identifiers[Constants.ODP.keyForVuid] = _vuid
        }
```

File: Sources/ODP/OdpVuidManager.swift (hunk 1 above)
Comment (muzahidul-opti) [reply Sources/ODP/OdpVuidManager.swift:24]: Declared singleton and add initialize method.
```
static let shared = VuidManager()
    
    func initialize(enabled: Bool) {
        self.enabled = enabled
        if enabled {
            self._vuid = load()
        } else {
            self.remove()
        }
    }
    
```

**Review by jaeopt — APPROVED** [review]
LGTM!

**Pull request merged**